clusterDeployer.py

```diff
@@ -445,6 +447,9 @@ def ensure_linked_to_bridge(self, lh: host.Host) -> None:
             logger.info(f"Incorrect master set for interface {api_network}")
             sys.exit(-1)
 
+        logger.info(f"Setting interface {api_network} as unmanaged in NetworkManager")
+        lh.run(f"nmcli device set {api_network} managed no")
```

Comment:
This will give issues if there are multiple clusters on the same system. We need some kind of ref counting, or a file that we write that contains all deployed clusters so that cleanup is implemented properly.

Reply:
Is this a supported use case (multiple clusters on the same system, using the same physical interface)?

I think we have problems regardless of this change, on teardown [we remove the interface from the virtual bridge already today](https://github.com/bn222/cluster-deployment-automation/blob/7e12a5295e07848325330e6bda2f4edbfdf2b8a1/clusterDeployer.py#L377).

Reply:
yes it is, and it's an important one. Two sets of masters for two different clusters using the same physical interface on the same node.

Reply:
Good point about that line, this is indeed a bug

Reply:
Hmm, but that use case is broken from day-1 AFAICT because teardown of cluster1 will remove the interface from virbr.  If I'm not wrong that's the case since ce06c61b4d18 ("Add automation for x86 and bf")

Reply:
correct. I'll merge it. In most cases, when there are two clusters deployed on a node, you tend to re-add one of the clusters after removing it anyway so this gets resolved in any case when that happens.